src/hooks/core/useCoreChatSystemHook.ts

```diff
@@ -136,12 +144,50 @@ const useCoreChatSystemHook = () => {
     return websocketData;
   };
 
+  const helperWebsocketForDeletedMessage = async (websocketData: GetstreamWebsocket) => {
+    const websocketMessage = websocketData?.message;
+    const isDeletedMessage = websocketMessage?.message_type === MESSAGE_TYPE_DELETED;
+
+    if (isDeletedMessage) {
+      const selectedChat = await ChatSchema.getByid(
+        localDb,
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        websocketMessage.deleted_message_id!
+      );
+      const selectedCreatedAt = selectedChat.createdAt;
+      const selectedUpdatedAt = selectedChat.updatedAt;
+
+      websocketMessage.id = websocketMessage.deleted_message_id ?? '';
+      websocketMessage.text = DELETED_MESSAGE_TEXT;
+      websocketMessage.created_at = selectedCreatedAt ?? websocketMessage.created_at;
+      websocketMessage.updated_at = selectedUpdatedAt ?? websocketMessage.updated_at;
+
+      ChatSchema.updateDeletedRepliedChat(
+        localDb,
+        websocketData?.channel_id,
+        websocketMessage.deleted_message_id ?? '',
+        selectedChat.createdAt
+      );
+
+      const {replyTarget} = replyPreview;
+      if (replyTarget && replyTarget?.id === websocketMessage?.deleted_message_id) {
+        const newReplyPreview = {...replyTarget};
+        newReplyPreview.message = DELETED_MESSAGE_TEXT;
+        newReplyPreview.message_type = MESSAGE_TYPE_DELETED;
+        setReplyTarget(newReplyPreview, dispatch);
+      }
+    }
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This function `helperWebsocketForDeletedMessage` seems to be doing a lot of things - fetching data from the database, updating the database, and also updating the state. This violates the Single Responsibility Principle. Consider breaking this function into smaller functions each with a single responsibility.



<!-- This is an auto-generated comment by OSS CodeRabbit -->